types/datum.go

```diff
@@ -194,7 +194,10 @@ var sink = func(s string) {
 
 // GetBytes gets bytes value.
 func (d *Datum) GetBytes() []byte {
-	return d.b
+	if d.b != nil {
```

Comment:
This change is necessary, because

nil and []byte{} are different.

In `ParseWithParams`, nil get `NULL` while []byte{} get '' ... 

This change is for here,  when `GetUpbound()` returns `nil, it need a '' instead of a NULL:
https://github.com/pingcap/tidb/pull/22636/files#diff-d259e06c930553fe055c26b2a44cac2eaf0daabee4d6a3b568273b272de3bc55R760